plugins/inputs/postgresql/service.go

```diff
@@ -111,33 +110,16 @@ func (p *Service) Start(telegraf.Accumulator) (err error) {
 	// Specific support to make it work with PgBouncer too
 	// See https://github.com/influxdata/telegraf/issues/3253#issuecomment-357505343
 	if p.IsPgBouncer {
-		d := &stdlib.DriverConfig{
-			ConnConfig: pgx.ConnConfig{
-				PreferSimpleProtocol: true,
-				RuntimeParams: map[string]string{
-					"client_encoding": "UTF8",
-				},
-				CustomConnInfo: func(c *pgx.Conn) (*pgtype.ConnInfo, error) {
-					info := c.ConnInfo.DeepCopy()
-					info.RegisterDataType(pgtype.DataType{
-						Value: &pgtype.OIDValue{},
-						Name:  "int8OID",
-						OID:   pgtype.Int8OID,
-					})
-					// Newer versions of pgbouncer need this defined. See the discussion here:
-					// https://github.com/jackc/pgx/issues/649
-					info.RegisterDataType(pgtype.DataType{
-						Value: &pgtype.OIDValue{},
-						Name:  "numericOID",
-						OID:   pgtype.NumericOID,
-					})
```

Comment:
it looks like we're still losing support for these data types? Do you know that they're both covered by the new v4 library?

Reply:
The mapping of the [data types](https://github.com/jackc/pgtype/blob/b26cd223783bf410ec9cd789ae872cef56610d93/pgtype.go#L65) is done inside the driver. Within pgx v4 the mapping has been outsourced and there is a separate repository for it. The version of the driver I am using, uses 1.3.0 of the outsourced types repositories. Furthermore, I have also made the corresponding integration test functional and could already test the corresponding extraction of the metrics there.

Reply:
got it. just trying to make sure we didn't inadvertently remove support for these two types. Otherwise we're good to go

Reply:
Thanks for the link to the pgtype mapping. I didn't know that's how it works but it makes sense now.